love/components/answers/lover-answers.tsx

```diff
@@ -6,21 +6,40 @@ import { partition } from 'lodash'
 import { Col } from 'web/components/layout/col'
 import { FreeResponseDisplay } from './free-response-display'
 import { OpinionScale } from './opinion-scale-display'
+import { useUserAnswersAndQuestions } from 'love/hooks/use-questions'
+import { useState } from 'react'
 
 export function LoverAnswers(props: {
   isCurrentUser: boolean
-  answers: rowFor<'love_answers'>[]
   router: NextRouter
-  questions: rowFor<'love_questions'>[]
   user: User
 }) {
-  const { isCurrentUser, answers, router, questions, user } = props
+  const { isCurrentUser, router, user } = props
+
+  const [refresh, setRefresh] = useState<number>(0)
+
+  const { questions: allQuestions, answers: allAnswers } =
+    useUserAnswersAndQuestions(user?.id, refresh)
+
+  const answers = allAnswers.filter(
+    (a) => a.multiple_choice != null || a.free_response || a.integer
+  )
+
+  const answerQuestionIds = new Set(answers.map((answer) => answer.question_id))
+
+  const questions = allQuestions.filter((question) =>
+    answerQuestionIds.has(question.id)
+  )
 
   const [multiChoiceAnswers, otherAnswers] = partition(
     answers,
     (a) => a.multiple_choice != null
   )
 
+  function refreshAnswers() {
```

Comment:
This is not the right way to do this. The useUserAnswersAndQuestions should return a `refreshResults` callback in addition to the `questions` and `answers` it returns now. Then anyone can call the refresh callback on demand. With your current setup, anyone who wants to refresh their results has to create this extra scaffolding of a refresh state and an incrementing index.